StoreTestWPF.ViewModel/ViewModels/CakeViewModel.cs

```diff
@@ -0,0 +1,90 @@
+﻿using StoreTestWPF.DAL.Models;
+using System;
+
+namespace StoreTestWPF.ViewModel.ViewModels
+{
+    public class CakeViewModel : ViewModelBase
+    {
+        private Cake cake;
+
+        public CakeViewModel()
+        {
+        }
+
+        public CakeViewModel(Cake cake)
+        {
+            if (cake == null)
+            {
+                throw new NullReferenceException();
```

Comment:
In such cases we should throw more informative message and more specific exception.
ArgumentNullException(nameof(cake));

Reply:
Fixed